File: Mahek Jhamb/Day 4/4B.py
class Solution:
    def myPow(self, x: float, n: int) -> float:
        def recurPow(x,n):
            n=abs(n)
            if n==0:
                return 1
            elif n%2==0:
                return recurPow(x*x,n//2)
            else:
                return x*recurPow(x*x,(n-1)//2)
        ans=recurPow(x,n)
        return float(ans) if n>=0 else 1/ans
        

File: Tanmay Jaiswal/day-4/4b.cpp
class Solution {
public:
    double myPow(double x, int n) {
        double res = 1.0;
        int N = n;

        n = abs(n);
        while (n>0) {
            if (n&1) res *= x;
            x *= x;
            n >>= 1;
        }

        if (N>=0) return res;
        return 1.0/res;
    }
};
